File: backend/app/auth/auth_service.py
from sqlalchemy.orm import Session

from app.database.models import User

from app.auth.auth_utils import (
    hash_password,
    verify_password,
    create_access_token
)


def register_user(
    db: Session,
    name,
    email,
    password
):

    existing_user = (
        db.query(User)
        .filter(
            User.email == email
        )
        .first()
    )

    if existing_user:

        raise Exception(
            "Email already exists"
        )

    user = User(
        name=name,
        email=email,
        password=hash_password(
            password
        )
    )

    db.add(user)

    db.commit()

    db.refresh(user)

    return user


def login_user(
    db: Session,
    email,
    password
):

    user = (
        db.query(User)
        .filter(
            User.email == email
        )
        .first()
    )

    if not user:

        return None

    if not verify_password(
        password,
        user.password
    ):

        return None

    token = create_access_token(
        {
            "user_id": user.id
        }
    )

    return token

File: frontend/src/services/authService.ts
import axios from "axios";

const API = "http://localhost:8000";

export const loginUser = async (
  email: string,
  password: string
) => {
  const response = await axios.post(
    `${API}/auth/login`,
    {
      email,
      password,
    }
  );

  return response.data;
};

export const registerUser = async (
  name: string,
  email: string,
  password: string
) => {
  const response = await axios.post(
    `${API}/auth/register`,
    {
      name,
      email,
      password,
    }
  );

  return response.data;
};